File: archives/backend-python-abandonne/agents/welcome_agent.py
"""
Agent d'accueil (welcomeAgent)
Accueille l'utilisateur nommément si son prénom est connu.
"""

from datetime import datetime


def run(context: dict) -> dict:
    """
    context attendu (clés utilisées si présentes) :
      - user_name: str | None
    Retourne un dict ajouté au contexte partagé de l'orchestrateur,
    et un message lisible ajouté à context["messages"].
    """
    user_name = context.get("user_name")
    hour = datetime.now().hour

    if 5 <= hour < 12:
        moment = "Bonjour"
    elif 12 <= hour < 18:
        moment = "Bon après-midi"
    else:
        moment = "Bonsoir"

    if user_name:
        message = f"{moment} {user_name}, bienvenue sur Azuska Z."
    else:
        message = f"{moment}, bienvenue sur Azuska Z."

    result = {
        "agent": "welcome",
        "status": "success",
        "greeting": message,
    }

    context.setdefault("messages", []).append(
        {"agent": "welcome", "content": message}
    )

    return result

File: frontend/lib/agents/welcomeAgent.ts
// lib/agents/welcomeAgent.ts
import { Agent, AgentResult, SharedContext } from "../../core/agentTypes";

export function moment(hour: number = new Date().getHours()): string {
  if (hour >= 5 && hour < 12) return "Bonjour";
  if (hour >= 12 && hour < 18) return "Bon après-midi";
  return "Bonsoir";
}

export const welcomeAgent: Agent = {
  run(context: SharedContext): AgentResult {
    const greeting = context.userName
      ? `${moment()} ${context.userName}, bienvenue sur Azuska Z.`
      : `${moment()}, bienvenue sur Azuska Z.`;

    context.messages.push({ agent: "welcome", content: greeting });

    return { agent: "welcome", status: "success", greeting };
  },
};
